runtime/jvmti/jvmtiGeneral.c

```diff
@@ -188,12 +185,9 @@ jvmtiGetVersionNumber(jvmtiEnv* env,
 	ENSURE_NON_NULL(version_ptr);
 
 #if JAVA_SPEC_VERSION >= 11
-#if JAVA_SPEC_VERSION >= 15
-	if (J2SE_VERSION(vm) >= J2SE_V15) {
+	if (JAVA_SPEC_VERSION >= 15) {
 		rv_version = JVMTI_VERSION_15;
-	} else
-#endif /* JAVA_SPEC_VERSION >= 15 */
-	if (J2SE_VERSION(vm) >= J2SE_V11) {
+	} else {
 		rv_version = JVMTI_VERSION_11;
 	}
```

Comment:
This doesn't compile for jdk11 (for which `JVMTI_VERSION_15` is not defined); I suggest
```
#if JAVA_SPEC_VERSION >= 15
	jint rv_version = JVMTI_VERSION_15;
#elif JAVA_SPEC_VERSION >= 11 /* JAVA_SPEC_VERSION >= 15 */
	jint rv_version = JVMTI_VERSION_11;
#else /* JAVA_SPEC_VERSION >= 15 */
	jint rv_version = JVMTI_1_2_3_SPEC_VERSION;
#endif /* JAVA_SPEC_VERSION >= 15 */
```

Reply:
Updated with slight modification.